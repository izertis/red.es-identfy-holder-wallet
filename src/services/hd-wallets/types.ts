import { derivationPathFormats } from './constants'

type randomType = 'random'

export type settingType = {
	m?: undefined
	Z?: string
	R?: string | randomType
	SSSSS?: [string, string, string, string, string] | randomType
	W?: string | randomType
	M?: string
	T?: string
	N?: string
	B?: randomType
	BB?: [string, string] | randomType
}
export type validKeysType = keyof settingType

export type derivationPathFormat =
	| derivationPathFormats.MAIN_IDENTITY
	| derivationPathFormats.NETWORK_DID
	| derivationPathFormats.INTERACTING_DID
