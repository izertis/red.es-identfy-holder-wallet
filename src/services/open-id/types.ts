import { Jwt, JwtHeader, JwtPayload } from 'jsonwebtoken'
import { ModalPropsWithoutIsVisible } from '../../context/Modal.context'

export type ModalType = {
	showModal?: (props?: ModalPropsWithoutIsVisible) => void
	t?: (s: string) => string
	showMessage?: (x: { content: string; type: string }) => void
}

export type Grants = {
	authorization_code: {
		issuer_state: string
	}
	'urn:ietf:params:oauth:grant-type:pre-authorized_code'?: {
		'pre-authorized_code': string
		user_pin_required: boolean
	}
}

export type PreAuthorizeResponse = {
	access_token: string
	token_type: string
	expires_in: number
	id_token: string
	c_nonce: string
	c_nonce_expires_in: number
}

export type CredentialOffer = {
	credential_issuer: string
	credentials: {
		format: string
		types: string[]
		trust_framework: {
			name: string
			type: string
			uri: string
		}
	}[]
	grants: Grants
}

export type PresentationOffer = {
	aud: string
	client_id: string
	exp: number
	iat: number
	iss: string
	nonce: string
	presentation_definition: PresentationDefinition
	redirect_uri: string
	response_mode: string
	response_type: string
	scope: string
	state?: string
}

export type OpenIdCredentialIssuer = {
	credential_issuer: string
	authorization_server: string
	credential_endpoint: string
	deferred_credential_endpoint: string
	credentials_supported: {
		format: string
		types: string[]
		trust_framework: {
			name: string
			type: string
			uri: string
		}
		display: {
			name: string
			locale: string
		}[]
	}[]
}

export type OpenIdConfiguration = {
	redirect_uris: string[]
	issuer: string
	authorization_endpoint: string
	token_endpoint: string
	jwks_uri: string
	scopes_supported: string[]
	response_types_supported: string[]
	response_modes_supported: string[]
	grant_types_supported: string[]
	subject_types_supported: string[]
	id_token_signing_alg_values_supported: string[]
	request_object_signing_alg_values_supported: string[]
	request_parameter_supported: boolean
	request_uri_parameter_supported: boolean
	token_endpoint_auth_methods_supported: string[]
	request_authentication_methods_supported: {
		[key: string]: string[]
	}
	vp_formats_supported: {
		[key: string]: {
			alg_values_supported: string[]
		}
	}
	subject_syntax_types_supported: string[]
	subject_syntax_types_discriminations: string[]
	subject_trust_frameworks_supported: string[]
	id_token_types_supported: string[]
}

export type CredentialResponse = {
	format: 'jwt_vc'
	credential: string
	acceptance_token: string
}

export type AuthorizationRequest = {
	response_type: 'code'
	client_id: string
	redirect_uri: string
	scope: string
	issuer_state?: string
	state?: string
	authorization_details: string
	nonce?: string
	code_challenge?: string
	code_challenge_method?: 'S256' | 'plain'
	client_metadata?: string
}

export type HolderServiceWalletMetadata = {
	authorization_endpoint: string
	response_types_supported: string[]
	vp_formats_supported: {
		jwt_vp: {
			alg_values_supported: string[]
		}
		jwt_vc: {
			alg_values_supported: string[]
		}
	}
	scopes_supported: string[]
	subject_types_supported: string[]
	id_token_signing_alg_values_supported: string[]
	request_object_signing_alg_values_supported: string[]
	subject_syntax_types_supported: string[]
	id_token_types_supported: string[]
}

export type AuthorizationDetail = {
	type: 'openid_credential'
	locations?: string[]
	format: 'jwt_vp' | 'jwt_vc' | string
	types: string[]
}

export type AuthorizationResponse = {
	code: string
	state?: string
}

export type GrantType = 'urn:ietf:params:oauth:grant-type:pre-authorized_code' | 'preauthorized_code'

export type TokenRequest = {
	redirect_uri: string
	grant_type: GrantType
	client_id?: string
	code?: string
	client_assertion_type?: boolean
	code_verifier?: string
	user_pin?: string
	'pre-authorized_code'?: string
}

export type TokenResponse = {
	access_token: string
	refresh_token: string
	token_type: string
	expires_in: number
	id_token: string
	c_nonce: string
	c_nonce_expires_in: number
}

export type VPTokenRequestPayload = {
	iss: string
	aud: string
	exp: number
	response_type: string
	response_mode: string
	client_id: string
	redirect_uri: string
	scope: string
	nonce: string
	presentation_definition_uri: string
}

export type AuthorizationRequestResponse = {
	client_id: string
	key: string
	nonce: string
	redirect_uri: string
	request_uri: string
	response_mode: string
	response_type: string
	scope: string
	presentation_definition: PresentationDefinition
	presentation_definition_uri: string
	request: string
	error: string
	error_description: string
}

export type DeferredCredential = {
	credential_endpoint: string
	acceptance_token: string
}

export type PresentationDefinition = {
	id: string
	format: Format
	input_descriptors: InputDescriptor[]
	purpose?: string
}

type InputDescriptor = {
	id: string
	name?: string
	format?: Format
	constraints?: {
		fields: FieldConstraint[]
	}
}

type FieldConstraint = {
	path: string[]
	filter: {
		type: string
		contains: {
			const: string
		}
	}
}

export type Format = {
	jwt_vc?: {
		alg: string[]
	}
	jwt_vp?: {
		alg: string[]
	}
	jwt_vc_json?: {
		alg: string[]
	}
	jwt_vp_json?: {
		alg: string[]
	}
}

export type TrustedFramework = 'lacchain' | 'alastria' | 'ebsi'

export type CredentialData = {
	id: string
	issuer: string
	validFrom: any
	expirationDate: any
	timestamp?: Date
	credential: Credential
	status: 'Active' | 'Revoked'
}

// CREDENTIALS SCHEMA
export type Vc = {
	'@context': string[]
	id: string
	type: string[]
	credentialSchema: {
		id: string
		type: string
	}
	credentialSubject: {
		id: string
	}
	validFrom?: string // Date timestamp. Example: "2010-01-01T19:23:24Z",
	validUntil?: string // Date timestamp. Example: "2010-01-01T19:23:24Z",
	expirationDate?: string // Date timestamp. Example: "2010-01-01T19:23:24Z",
	issuanceDate: string
	issuer: string
	issued?: string
	credentialStatus?: W3CCredentialStatus
	proof?: EmbeddedProof
	[x: string]: any
}

export interface EmbeddedProof {
	id?: string
	type: string
	proofPurpose: string
	verificationMethod: string
	created?: string // Date timestamp. Example: "2010-01-01T19:23:24Z",
	expires?: string // Date timestamp. Example: "2010-01-01T19:23:24Z",
	domain?: string
	challenge?: string
	proofValue: string
	previousProof?: string | string[]
	nonce?: string
}

export interface W3CCredentialStatus {
	id?: string
	type: string
	[key: string]: any
}

export interface StatusListEntry2021 {
	id: string
	type: 'StatusList2021Entry'
	statusPurpose: 'revocation' | 'suspension'
	statusListIndex: string
	statusListCredential: string
}

export interface StatusListCredentialData {
	id: string
	type: 'StatusList2021'
	statusPurpose: 'revocation' | 'suspension'
	encodedList: string
}

export type Credential = {
	header: JwtHeader
	payload: JwtPayload & {
		vc: Vc
	}
}

export type chosenCredential = {
	format: 'jwt_vc' | 'ldp_vc'
	trust_framework: { name: string; type: string; uri: string }
	types: ['VerifiableCredential', 'VerifiableAttestation', string]
}

export type PresentationData = {
	id: string
	issuer: string
	validFrom: any
	expirationDate: any
	timestamp: Date
	credential: Credential
	status: 'Active' | 'Revoked'
}

export type ActionLog = {
	id: string
	actionType: 'Credential added' | 'Credential expired' | 'Presentation emitted' | 'Presentation revoked'
	issuer: string
	date: Date | string
	data?: any
}
