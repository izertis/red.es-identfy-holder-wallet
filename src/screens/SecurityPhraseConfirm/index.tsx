import { useContext, useEffect, useState } from 'react'
import { useTranslation } from 'react-i18next'
import { RootStackScreenProps } from '../../../types'
import Button from '../../components/atomic_components/Button'
import InputStyled from '../../components/atomic_components/InputStyled'
import SeedButton from '../../components/SeedButton'
import { SCREEN } from '../../constants/screens'
import { MessageContext } from '../../context/UserMessage.context'
import SecurityPhraseConfirmI18nKeys from './i18n/keys'
import SecurityPhraseConfirmStyled from './styles'
import { shuffleArray } from '../../utils/shuffleArray'

const SecurityPhraseConfirm = ({ route, navigation }: RootStackScreenProps<'SecurityPhraseConfirm'>) => {
	const { t } = useTranslation(SCREEN.SecurityPhraseConfirm)

	const { mneumonic, derivationPath }: any = route.params

	const {  showMessage } = useContext(MessageContext)
	const [seedPhrase, setSeedPhrase] = useState<string[]>((mneumonic.split(' ')))
	const [showedPhrase, setShowedPhrase] = useState<string[]>([])
	const [originalSeed, setOriginalSeed] = useState<string[]>(mneumonic.split(' '))

	useEffect(() => {
		resetSeeds()
	}, [])

	const handlePress = (word: string) => {
    setShowedPhrase([...showedPhrase, word])
    setSeedPhrase(
      seedPhrase.filter(
        (i, index, arr) =>
          i !== word || index !== arr.findIndex((w) => w === word)
      )
    )
  }

	const handleCheck = async () => {
		if (JSON.stringify(originalSeed) === JSON.stringify(showedPhrase)) {
			navigation.navigate('DerivationPath', { derivationPath: derivationPath })
			resetSeeds()
		} else {
			const content = t(SecurityPhraseConfirmI18nKeys.ERROR)
			showMessage({ content, type: 'error' })
		}
	}
	const resetSeeds = () => {
		setSeedPhrase(shuffleArray(mneumonic.split(' ')))
		setOriginalSeed(mneumonic.split(' '))
		setShowedPhrase([])
	}

	return (
		<SecurityPhraseConfirmStyled.MainContainer>
			<SecurityPhraseConfirmStyled.Container>
				<SecurityPhraseConfirmStyled.RecoveryButton onPress={() => navigation.pop()}>
					{t(SecurityPhraseConfirmI18nKeys.GO_BACK)}
				</SecurityPhraseConfirmStyled.RecoveryButton>
				<SecurityPhraseConfirmStyled.Title testID={'title'}>
					{t(SecurityPhraseConfirmI18nKeys.CONFIRM_TITLE)}
				</SecurityPhraseConfirmStyled.Title>
				<InputStyled
					style={{ height: 120 }}
					editable={false}
					multiline
					label={t(SecurityPhraseConfirmI18nKeys.INPUT_LABEL)}
					value={showedPhrase.join(' ')}
					placeholder={t(SecurityPhraseConfirmI18nKeys.INPUT_PLACEHOLDER)}
				></InputStyled>

				<SecurityPhraseConfirmStyled.ButtonContainer>
					{seedPhrase.map((word, index) => {
						return (
							<SeedButton
								text={word}
								key={`SeedButton-${word}-${index}`}
								onPress={() => {
									handlePress(word)
								}}
							/>
						)
					})}
				</SecurityPhraseConfirmStyled.ButtonContainer>
				<SecurityPhraseConfirmStyled.ResetButton
					disabled={showedPhrase.length < 1}
					onPress={() => {
						resetSeeds()
					}}
				>
					{t(SecurityPhraseConfirmI18nKeys.RESET_BUTTON)}
				</SecurityPhraseConfirmStyled.ResetButton>

				<SecurityPhraseConfirmStyled.ContainerBottom>
					<Button
						disabled={showedPhrase.length < 12}
						onPress={() => {
							handleCheck()
						}}
					>
						{t(SecurityPhraseConfirmI18nKeys.CONFIRM_BUTTON)}
					</Button>
				</SecurityPhraseConfirmStyled.ContainerBottom>
			</SecurityPhraseConfirmStyled.Container>
		</SecurityPhraseConfirmStyled.MainContainer>
	)
}

export default SecurityPhraseConfirm
