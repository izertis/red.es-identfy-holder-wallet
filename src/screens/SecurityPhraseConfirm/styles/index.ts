import defaultStyles from "./defaultStyles"
import proyectSyles from "./proyectStyles"

const SecurityPhraseConfirmStyled = {
  ...defaultStyles,
  ...proyectSyles
}

export default SecurityPhraseConfirmStyled
