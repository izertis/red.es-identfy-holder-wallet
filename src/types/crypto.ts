export type JWT = `ey${string}.ey${string}.${string}`