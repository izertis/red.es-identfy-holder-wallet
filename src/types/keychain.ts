import { ActionLog, CredentialData, DeferredCredential, PresentationData } from '../services/open-id/types'

export type KeyChainData = {
	encryptedPrivateKey?: string | ArrayBuffer
	did?: { lacchain?: string; alastria?: string; ebsi?: string }
	JWKKeys?: { privateKeyJWK: string; publicKeyJWK: string; privateKeyHex: string; publicKeyHex: string }
	signatureKeyPair?: { publicKey: string; privateKey: string }
	credentials?: CredentialData[]
	presentations?: PresentationData[]
	deferred_credentials?: DeferredCredential[]
	actionLogs?: ActionLog[]
}
