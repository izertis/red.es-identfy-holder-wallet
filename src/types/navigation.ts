import { MaterialTopTabScreenProps } from '@react-navigation/material-top-tabs'
import { NativeStackNavigationOptions } from '@react-navigation/native-stack'

export type navigationScreensConfigType = {
	name: string
	component: (props: any) => JSX.Element
	options:
		| NativeStackNavigationOptions
		| MaterialTopTabScreenProps<{}>
		| ((x: any) => NativeStackNavigationOptions | MaterialTopTabScreenProps<{}>)
}
