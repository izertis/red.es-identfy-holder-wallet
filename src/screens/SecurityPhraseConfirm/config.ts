import i18next from '../../../i18n.config'
import { SCREEN } from "../../constants/screens"
import { navigationScreensConfigType } from "../../types/navigation"
import es from "./i18n/es"
import SecurityPhraseConfirm from '.'

const ScreenName = SCREEN.SecurityPhraseConfirm

i18next.addResourceBundle("es", ScreenName, es)

const SecurityPhraseConfirmConfig: navigationScreensConfigType = {
  name: ScreenName,
  component: SecurityPhraseConfirm,
  options: { headerShown: false },
}

export default SecurityPhraseConfirmConfig
