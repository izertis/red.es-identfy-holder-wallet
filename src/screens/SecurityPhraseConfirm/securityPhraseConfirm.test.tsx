import { RouteProp } from '@react-navigation/native'
import { NativeStackNavigationProp } from '@react-navigation/native-stack'
import { act, render } from '@testing-library/react-native'
import SecurityPhraseConfirm from '.'
import { RootStackParamList } from '../../../types'

type ScreenNavigationProp = NativeStackNavigationProp<RootStackParamList, 'SecurityPhraseConfirm'>

type SecurityPhraseConfirmRouteProp = RouteProp<RootStackParamList, 'SecurityPhraseConfirm'>

const mockedNavigation: ScreenNavigationProp | any = {
	navigate: jest.fn(),
}

const mockedRoute: SecurityPhraseConfirmRouteProp = {
	key: 'mocked-key',
	name: 'SecurityPhraseConfirm',
	params: {
		mneumonic: 'mocked-mnemonic',
		derivationPath: 'mocked-derivation-path',
	},
}

describe('SecurityPhraseConfirm screen', () => {
	it('should render SecurityPhraseConfirm screen', () => {
		const result = render(<SecurityPhraseConfirm navigation={mockedNavigation} route={mockedRoute} />)

		expect(result.toJSON()).toMatchSnapshot()
	})
	it('Should renders items correctly', () => {
		const { getAllByRole, getByTestId } = render(
			<SecurityPhraseConfirm navigation={mockedNavigation} route={mockedRoute} />
		)

		const title = getByTestId('title')
		const confirmBtn = getAllByRole('button')

		expect(title).toBeDefined()
		expect(confirmBtn).toBeDefined()
	})
})
