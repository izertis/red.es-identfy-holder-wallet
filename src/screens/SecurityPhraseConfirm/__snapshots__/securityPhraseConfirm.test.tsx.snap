// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SecurityPhraseConfirm screen should render SecurityPhraseConfirm screen 1`] = `
<View
  className="StyledNativeComponent sc-gKPRtg ioznsx"
>
  <View
    className="sc-ipEyDJ iaMRWP"
  >
    <View
      collapsable={false}
      style={
        {
          "backgroundColor": "transparent",
          "borderRadius": 20,
          "shadowColor": "#000",
          "shadowOffset": {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 0,
          "shadowRadius": 0,
        }
      }
      testID="button-container-outer-layer"
    >
      <View
        className="sc-iBYQkv ecSfdn"
        collapsable={false}
        style={
          {
            "backgroundColor": "transparent",
            "borderColor": "transparent",
            "borderRadius": 20,
            "borderStyle": "solid",
            "borderWidth": 0,
            "flex": undefined,
            "minWidth": 64,
            "shadowColor": "#000",
            "shadowOffset": {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 0,
            "shadowRadius": 0,
          }
        }
        testID="button-container"
      >
        <View
          accessibilityRole="button"
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": false,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            [
              {
                "overflow": "hidden",
              },
              {
                "borderRadius": 20,
              },
            ]
          }
          testID="button"
        >
          <View
            style={
              [
                {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                },
                undefined,
              ]
            }
          >
            <View
              style={
                [
                  {
                    "marginLeft": 12,
                    "marginRight": -4,
                  },
                  {
                    "marginLeft": 16,
                    "marginRight": -16,
                  },
                  {
                    "marginLeft": 12,
                    "marginRight": -8,
                  },
                ]
              }
              testID="button-icon-container"
            >
              <Text />
            </View>
            <Text
              numberOfLines={1}
              selectable={false}
              style={
                [
                  {
                    "textAlign": "left",
                  },
                  {
                    "color": "rgba(28, 27, 31, 1)",
                    "writingDirection": "ltr",
                  },
                  [
                    {
                      "fontFamily": "System",
                      "fontSize": 14,
                      "fontWeight": "500",
                      "letterSpacing": 0.1,
                      "lineHeight": 20,
                    },
                    [
                      {
                        "marginHorizontal": 16,
                        "marginVertical": 9,
                        "textAlign": "center",
                      },
                      false,
                      {
                        "marginHorizontal": 16,
                      },
                      undefined,
                      false,
                      {
                        "color": "rgba(103, 80, 164, 1)",
                        "fontFamily": "System",
                        "fontSize": 14,
                        "fontWeight": "500",
                        "letterSpacing": 0.1,
                        "lineHeight": 20,
                      },
                      {
                        "color": "#eb5b5b",
                      },
                    ],
                  ],
                ]
              }
              testID="button-text"
            >
              GO_BACK
            </Text>
          </View>
        </View>
      </View>
    </View>
    <Text
      bold={true}
      className="sc-hLBbgP sc-jSUZER sc-ftTHYK PgSAu dXSLfz ejyDuZ"
      fontFamily="Montserrat-Bold"
      style={
        [
          {
            "textAlign": "left",
          },
          {
            "color": "rgba(28, 27, 31, 1)",
            "fontFamily": "System",
            "fontWeight": "400",
            "letterSpacing": 0,
          },
          {
            "writingDirection": "ltr",
          },
          undefined,
        ]
      }
      testID="title"
    >
      CONFIRM_TITLE
    </Text>
    <View
      style={
        [
          {
            "backgroundColor": "#fff",
          },
          undefined,
        ]
      }
    >
      <View
        style={{}}
      >
        <View
          pointerEvents="none"
          style={
            [
              {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 6,
              },
              {
                "backgroundColor": "rgba(255, 251, 254, 1)",
                "borderColor": "#363d58",
                "borderRadius": 4,
                "borderWidth": 1,
              },
              undefined,
            ]
          }
          testID="text-input-outline"
        />
        <View>
          <View
            style={
              [
                {
                  "paddingBottom": 0,
                },
                {
                  "minHeight": 120,
                  "paddingTop": 8,
                },
              ]
            }
          >
            <View
              collapsable={false}
              pointerEvents="none"
              style={
                {
                  "bottom": 0,
                  "left": 0,
                  "opacity": 1,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                  "transform": [
                    {
                      "translateX": 0,
                    },
                  ],
                  "zIndex": 3,
                }
              }
            >
              <Text
                collapsable={false}
                maxFontSizeMultiplier={1.5}
                numberOfLines={1}
                style={
                  {
                    "backgroundColor": "rgba(255, 251, 254, 1)",
                    "color": "transparent",
                    "fontFamily": "System",
                    "fontSize": 16,
                    "fontWeight": undefined,
                    "left": 8,
                    "letterSpacing": 0.15,
                    "lineHeight": undefined,
                    "opacity": -0.6666666666666667,
                    "paddingHorizontal": 0,
                    "position": "absolute",
                    "textAlign": "left",
                    "top": 69,
                    "transform": [
                      {
                        "translateX": 0,
                      },
                      {
                        "translateY": 0,
                      },
                      {
                        "scale": 1,
                      },
                      {
                        "scaleY": 1,
                      },
                    ],
                    "width": -16,
                    "writingDirection": "ltr",
                  }
                }
              >
                INPUT_LABEL
              </Text>
              <Text
                collapsable={false}
                maxFontSizeMultiplier={1.5}
                numberOfLines={1}
                onLayout={[Function]}
                style={
                  {
                    "color": "#363d58",
                    "fontFamily": "System",
                    "fontSize": 16,
                    "fontWeight": undefined,
                    "left": 0,
                    "letterSpacing": 0.15,
                    "lineHeight": undefined,
                    "opacity": 0,
                    "paddingHorizontal": 16,
                    "position": "absolute",
                    "textAlign": "left",
                    "top": 68,
                    "transform": [
                      {
                        "translateX": 0,
                      },
                      {
                        "translateY": 0,
                      },
                      {
                        "scale": 1,
                      },
                    ],
                    "writingDirection": "ltr",
                  }
                }
                testID="text-input-outlined-label-active"
              >
                INPUT_LABEL
              </Text>
              <Text
                collapsable={false}
                maxFontSizeMultiplier={1.5}
                numberOfLines={1}
                style={
                  {
                    "color": "rgba(73, 69, 79, 1)",
                    "fontFamily": "System",
                    "fontSize": 16,
                    "fontWeight": undefined,
                    "left": 0,
                    "letterSpacing": 0.15,
                    "lineHeight": undefined,
                    "opacity": 0,
                    "paddingHorizontal": 16,
                    "position": "absolute",
                    "textAlign": "left",
                    "top": 68,
                    "transform": [
                      {
                        "translateX": 0,
                      },
                      {
                        "translateY": 0,
                      },
                      {
                        "scale": 1,
                      },
                    ],
                    "writingDirection": "ltr",
                  }
                }
                testID="text-input-outlined-label-inactive"
              >
                INPUT_LABEL
              </Text>
            </View>
            <TextInput
              className="sc-bcXHqe ibYIDo"
              containerStyle={
                {
                  "paddingHorizontal": 0,
                }
              }
              cursorColor="#363d58"
              editable={false}
              errorStyle={
                {
                  "color": "#eb5b5b",
                }
              }
              inputContainerStyle={
                {
                  "marginTop": 10,
                }
              }
              inputStyle={
                {
                  "fontSize": 16,
                }
              }
              maxFontSizeMultiplier={1.5}
              multiline={true}
              onBlur={[Function]}
              onChangeText={[Function]}
              onFocus={[Function]}
              placeholder=" "
              placeholderTextColor="#aaa"
              selectionColor="#eb5b5b"
              style={
                [
                  {
                    "margin": 0,
                    "zIndex": 1,
                  },
                  {
                    "height": 120,
                  },
                  {
                    "paddingBottom": 24,
                    "paddingTop": 24,
                  },
                  {
                    "color": "rgba(28, 27, 31, 1)",
                    "fontFamily": "System",
                    "fontSize": 16,
                    "fontWeight": undefined,
                    "letterSpacing": 0.15,
                    "lineHeight": undefined,
                    "paddingHorizontal": 16,
                    "textAlign": "left",
                    "textAlignVertical": "top",
                  },
                  false,
                  [
                    {},
                  ],
                  {
                    "fontFamily": "Muli",
                    "lineHeight": 25,
                  },
                ]
              }
              testID="text-input-outlined"
              underlineColorAndroid="transparent"
              value=""
            />
          </View>
        </View>
      </View>
    </View>
    <View
      className="sc-csuSiG bsbtvK"
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
      >
        <Text
          className="sc-dkrFOg fRBncy"
        >
          mocked-mnemonic
        </Text>
      </View>
    </View>
    <View
      collapsable={false}
      style={
        {
          "backgroundColor": "transparent",
          "borderRadius": 20,
          "shadowColor": "#000",
          "shadowOffset": {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 0,
          "shadowRadius": 0,
        }
      }
      testID="button-container-outer-layer"
    >
      <View
        borderWidth={0}
        className="sc-eDWCr kVuCHA"
        collapsable={false}
        color="#eb5b5b"
        style={
          {
            "backgroundColor": "transparent",
            "borderColor": "transparent",
            "borderRadius": 20,
            "borderStyle": "solid",
            "borderWidth": 0,
            "flex": undefined,
            "minWidth": 64,
            "shadowColor": "#000",
            "shadowOffset": {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 0,
            "shadowRadius": 0,
          }
        }
        testID="button-container"
      >
        <View
          accessibilityRole="button"
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": true,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            [
              {
                "overflow": "hidden",
              },
              {
                "borderRadius": 20,
              },
            ]
          }
          testID="button"
        >
          <View
            style={
              [
                {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                },
                undefined,
              ]
            }
          >
            <Text
              numberOfLines={1}
              selectable={false}
              style={
                [
                  {
                    "textAlign": "left",
                  },
                  {
                    "color": "rgba(28, 27, 31, 1)",
                    "writingDirection": "ltr",
                  },
                  [
                    {
                      "fontFamily": "System",
                      "fontSize": 14,
                      "fontWeight": "500",
                      "letterSpacing": 0.1,
                      "lineHeight": 20,
                    },
                    [
                      {
                        "marginHorizontal": 16,
                        "marginVertical": 9,
                        "textAlign": "center",
                      },
                      false,
                      {
                        "marginHorizontal": 12,
                      },
                      undefined,
                      false,
                      {
                        "color": "rgba(28, 27, 31, 0.38)",
                        "fontFamily": "System",
                        "fontSize": 14,
                        "fontWeight": "500",
                        "letterSpacing": 0.1,
                        "lineHeight": 20,
                      },
                      undefined,
                    ],
                  ],
                ]
              }
              testID="button-text"
            >
              RESET_BUTTON
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      className="sc-kDvujY ibcXlS"
    >
      <View
        collapsable={false}
        style={
          {
            "backgroundColor": "transparent",
            "borderRadius": 20,
            "shadowColor": "#000",
            "shadowOffset": {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 0,
            "shadowRadius": 0,
          }
        }
        testID="button-container-outer-layer"
      >
        <View
          collapsable={false}
          style={
            {
              "backgroundColor": "transparent",
              "borderColor": "transparent",
              "borderRadius": 20,
              "borderStyle": "solid",
              "borderWidth": 0,
              "flex": undefined,
              "minWidth": 64,
              "shadowColor": "#000",
              "shadowOffset": {
                "height": 0,
                "width": 0,
              },
              "shadowOpacity": 0,
              "shadowRadius": 0,
            }
          }
          testID="button-container"
        >
          <View
            accessibilityRole="button"
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": true,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              [
                {
                  "overflow": "hidden",
                },
                {
                  "borderRadius": 20,
                },
              ]
            }
            testID="button"
          >
            <View
              style={
                [
                  {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "center",
                  },
                  {
                    "backgroundColor": "#ccc",
                    "borderRadius": 15,
                    "fontFamily": "Muli",
                    "height": 39,
                    "width": 224,
                  },
                ]
              }
            >
              <Text
                numberOfLines={1}
                selectable={false}
                style={
                  [
                    {
                      "textAlign": "left",
                    },
                    {
                      "color": "rgba(28, 27, 31, 1)",
                      "writingDirection": "ltr",
                    },
                    [
                      {
                        "fontFamily": "System",
                        "fontSize": 14,
                        "fontWeight": "500",
                        "letterSpacing": 0.1,
                        "lineHeight": 20,
                      },
                      [
                        {
                          "marginHorizontal": 16,
                          "marginVertical": 9,
                          "textAlign": "center",
                        },
                        false,
                        {
                          "marginHorizontal": 12,
                        },
                        undefined,
                        false,
                        {
                          "color": "rgba(28, 27, 31, 0.38)",
                          "fontFamily": "System",
                          "fontSize": 14,
                          "fontWeight": "500",
                          "letterSpacing": 0.1,
                          "lineHeight": 20,
                        },
                        {
                          "color": "#fff",
                          "fontFamily": "Muli",
                          "letterSpacing": 0.57,
                        },
                      ],
                    ],
                  ]
                }
                testID="button-text"
              >
                CONFIRM_BUTTON
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
